kong/pdk/vault.lua

```diff
@@ -1455,7 +1455,9 @@ local function new(self)
 
     initialized = true
 
-    if self.configuration.database ~= "off" then
+    if self.configuration.database ~= "off" or  -- postgres
+       self.configuration.role == "data_plane" and self.configuration.cluster_incremental_sync -- incremental dp
```

Comment:
perhaps this long logic deserves a function that describes what we're checking, and maybe some round brackets to make it more clear.

Reply:
It is a style issue, so I will fix it later, not in this PR.

Reply:
done.